### PR TITLE
feat: support integration with experiment android

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary

 - Support integration with Experiment Android SDK with analytics connector

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?: no

### CODE REVIEW

**Review by liuyang1520 — APPROVED** [review]

File: android/src/main/java/com/amplitude/android/plugins/AnalyticsConnectorPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        connector.identityStore.editIdentity()
+            .updateUserProperties(userProperties)
+            .commit()
+
+        return event
+    }
+
+    companion object {
+        const val EXPOSURE_EVENT = "\$exposure"
+    }
```
Comment (bgiori) [inline android/src/main/java/com/amplitude/android/plugins/AnalyticsConnectorPlugin.kt:53]: This could be a top level definition outside the class

Comment (bgiori) [reply android/src/main/java/com/amplitude/android/plugins/AnalyticsConnectorPlugin.kt:53]: Companion objects are generally frowned upon in Kotlin I think

File: samples/kotlin-android-app/src/main/java/com/amplitude/android/sample/MainApplication.kt [hunk 2]
```diff
@@ -23,16 +27,27 @@ class MainApplication : Application() {
...
+            .build()
+
+        val experimentClient = Experiment.initializeWithAmplitudeAnalytics(this, EXPERIMENT_API_KEY, experimentConfig)
+
+        try {
+            experimentClient.fetch(null).get()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+
```
Comment (bgiori) [inline samples/kotlin-android-app/src/main/java/com/amplitude/android/sample/MainApplication.kt:42]: ❤️

**Review by bgiori — APPROVED** [review]

**Review by justin-fiedler — APPROVED** [review]

File: android/src/main/java/com/amplitude/android/plugins/AnalyticsConnectorPlugin.kt (hunk 1 above)
Comment (qingzhuozhen) [reply android/src/main/java/com/amplitude/android/plugins/AnalyticsConnectorPlugin.kt:53]: Interesting to know that. Didn't set up a constant class earlier, I could change that later. Was kind of under the wrong impression about it.

**Pull request merged**